src/MICore/CommandFactories/gdb.cs

```diff
@@ -194,5 +199,16 @@ public override Task EnableTargetAsyncOption()
             // that isn't actually supported by gdb. 
             await _debugger.CmdAsync("kill", ResultClass.None);
         }
+        public override async Task Signal(string sig)
+        {
+            string command = String.Format("-interpreter-exec console \"signal {0}\"", sig);
+            await _debugger.CmdAsync(command, ResultClass.running);
```

Comment:
I didn't use ConsoleCmdAsync because I couldn't give it a ResultClass.Running parameter (it expects Done). Signal will only be called when the process is stopped, so it might be better to add an optional second ResultClass parameter to ConsoleCmdAsync. However in this case there is no other output except the ResultClass to look for.